api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -546,6 +546,10 @@ message Cluster {
     // The http header to override destination address if :ref:`use_http_header <envoy_v3_api_field_config.cluster.v3.Cluster.OriginalDstLbConfig.use_http_header>`.
     // is set to true. If the value is empty, :ref:`x-envoy-original-dst-host <config_http_conn_man_headers_x-envoy-original-dst-host>` will be used.
     string http_header_name = 2;
+
+    // The port to override for the original dst address. This port
+    // will take precedence over filter state and header override ports
+    google.protobuf.UInt32Value upstream_port_override = 3;
```

Comment:
We can add validation `[(validate.rules).uint32 = {lte: 65535}]` just like https://github.com/envoyproxy/envoy/blob/667f0600743104857d59f56da4ab6aeea57fc60e/api/envoy/config/core/v3/address.proto#L78